plugins/Jobs.cpp

```diff
@@ -220,128 +238,192 @@ bool BedrockPlugin_Jobs::processCommand(SQLite& db, BedrockCommand& command) {
-                         result)) {
-                throw "502 Select failed";
+        vector<STable> jsonJobs;
+        if (SIEquals(request.methodLine, "CreateJob")) {
+            verifyAttributeSize(request, "name", 1, MAX_SIZE_SMALL);
+            jsonJobs.push_back(request.nameValueMap);
+        } else {
+            list<string> multipleJobs;
+            multipleJobs = SParseJSONArray(request["jobs"]);
+            for (auto& job : multipleJobs) {
+                jsonJobs.push_back(SParseJSONObject(job));
             }
+        }
 
-            // If we got a result, and it's data is the same as passed, we won't change anything.
-            if (!result.empty() && result[0][1] == request["data"]) {
-                SINFO("Job already existed with matching data, and unique flag was passed, reusing existing job "
-                      << result[0][0]);
-                content["jobID"] = result[0][0];
-                return true;
-            }
+        list<string> jobIDs;
+        for (auto& job : jsonJobs) {
+            // If unique flag was passed and the job exist in the DB, then we can finish the command without escalating to
+            // master.
+            uint64_t updateJobID = 0;
+            if (SContains(job, "unique")) {
+                SQResult result;
+                SINFO("Unique flag was passed, checking existing job with name " << job["name"]);
+                if (!db.read("SELECT jobID, data "
+                             "FROM jobs "
+                             "WHERE name=" + SQ(job["name"]) + ";",
+                             result)) {
+                    throw "502 Select failed";
+                }
 
-            // If we found a job, but the data was different, we'll need to update it.
-            if (!result.empty()) {
-                updateJobID = SToInt64(result[0][0]);
-            }
-        }
+                // If we got a result, and it's data is the same as passed, we won't change anything.
+                if (!result.empty() && result[0][1] == job["data"]) {
+                    SINFO("Job already existed with matching data, and unique flag was passed, reusing existing job "
+                          << result[0][0]);
 
-        // If no "firstRun" was provided, use right now
-        const string& safeFirstRun = request["firstRun"].empty() ? SCURRENT_TIMESTAMP() : SQ(request["firstRun"]);
+                    // If we are calling GetJob, return early, there are no more jobs to create.
+                    if(SIEquals(request.methodLine, "CreateJob")){
+                        content["jobID"] = result[0][0];
+                        return true;
+                    }
 
-        // If no data was provided, use an empty object
-        const string& safeData = request["data"].empty() ? SQ("{}") : SQ(request["data"]);
+                    // Append new jobID to list of created jobs.
+                    jobIDs.push_back(result[0][0]);
+                    continue;
```

Comment:
Seems like with this continue we'll skip updating the updateJobID variable below and thus we won't update the job. Am I reading this correctly?